userservice/src/main/java/com/gaurav/userservice/config/jwt/TokenProviderImpl.java

```diff
@@ -0,0 +1,73 @@
+package com.gaurav.userservice.config.jwt;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.ExpiredJwtException;
+import io.jsonwebtoken.JwtException;
+import io.jsonwebtoken.Jwts;
+import io.jsonwebtoken.security.Keys;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Configuration;
+
+import javax.crypto.SecretKey;
+import java.util.Base64;
+import java.util.Date;
+import java.util.Map;
+
+@Configuration
+public class TokenProviderImpl implements TokenProvider {
+
+    @Value("${jwt.secret}")
```

Comment:
Consider using a @ConfigurationProperties annotated class to centralize JWT properties, making them configurable and easy to manage.

`
@ConfigurationProperties(prefix = "jwt")
public class JwtProperties {
    private String secret;
    private long expiration;
    
    // getters and setters
}
`